galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistUtils.java

```diff
@@ -0,0 +1,214 @@
+		
+		if (WorklistQuery.runIdsInWorklist != null) {
+			
+			IRunResult run = null;
+			for (String runId : WorklistQuery.runIdsInWorklist) {
+				try {
+					run = getRunById(runId);
+				} catch (ResultArchiveStoreException e) {
+					throw new ServletException("Error while retrieving run for Run ID " + runId, e);
+				}
+				
+				if (run != null) {
+					
+					try {
+						
+						String runName = run.getTestStructure().getRunName();
+						String shortName = run.getTestStructure().getTestShortName();
+						String testClass = run.getTestStructure().getTestName();
+						String result = run.getTestStructure().getResult();
+						
+						WorklistItem worklistItem = new WorklistItem(runId, runName, shortName, testClass, result);
+						
+						runs.add(worklistItem);
+						
+					} catch (ResultArchiveStoreException e) {
+						throw new ServletException("Error while retriving Worklist information for Run ID " + runId, e);
+					}
+					
+				} else {
+					WorklistQuery.runIdsNotFound.add(runId);
+					try {
+						removeRunIdFromDss(runId);
+					} catch (Exception e) {
+						throw new ServletException("Error while updating the DSS", e);
+					}
+				}
+				
+			}
+		}
+		
+		return runs;
+	}
+	
+	protected IRunResult getRunById(String id) throws ResultArchiveStoreException {
+
+		IRunResult run = null;
+	       
+        for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
+           
+           run = directoryService.getRunById(id);
+           
+           if(run != null) {
+              break;
+           }
+           
+        }
+		
+		return run;
+	}
+	
+	protected void removeRunIdFromDss(String runId) throws DynamicStatusStoreException, FrameworkPropertyFileException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String oldValue = dss.get("user.username.worklist");
+		
+		if (oldValue != null) {
+			
+			JsonObject jsonObject = new JsonParser().parse(oldValue).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			JsonElement id = new JsonParser().parse(runId);
+			if (runIds.contains(id)) {
+				runIds.remove(id);
+				jsonObject.remove("runIds");
+				jsonObject.add("runIds", runIds);
+			}	
+			
+			String newValue = jsonObject.toString();
+			
+			dss.performActions(
+					new DssSwap("user.username.worklist", oldValue, newValue));
+			
+		}
+			
+	}
+	
+	protected void addRunIdToDss(String runId) throws DynamicStatusStoreException, FrameworkPropertyFileException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String oldValue = dss.get("user.username.worklist");
+		
+		if (oldValue != null) {
+			
+			JsonObject jsonObject = new JsonParser().parse(oldValue).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			JsonElement id = new JsonParser().parse(runId);
```

Comment:
as with the remove I believe this is unsafe code,  better to iterate and do a equals with getasstring